pkg/azure/actuator_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure_test
+
+import (
+	"testing"
+
+	"github.com/openshift/cloud-credential-operator/pkg/azure"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestAnnotations(t *testing.T) {
+	var tests = []struct {
+		in        corev1.Secret
+		errRegexp string
+	}{
+		{validRootSecret, ""},
+		{rootSecretBadAnnotation, "invalid mode"},
+		{rootSecretNoAnnotation, "cannot proceed without cloud cred secret annotation.*"},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			f := newFakeClient(t)
```

Comment:
i see the custom fake client you made, and i'm wondering why you didn't go with the fake client you get from "sigs.k8s.io/controller-runtime/pkg/client/fake"

Reply:
didn't know about that one. Thank you for pointing it out. I'll certainly use it again in the future.